core-web/libs/sdk/experiments/src/lib/parser/parser.ts

```diff
@@ -63,3 +63,122 @@ const verifyRegex = (regexToCheck: string | null, href: string): boolean => {
         return false;
     }
 };
+
+/**
+ * This function parses the fetched and stored experiment data and prepares it for storage.
```

Comment:
"This function parses the fetched and stored", should we split?

Reply:
```
 * @param { AssignedExperiments | null } fetchExperiments - The experiment data fetched from the API.
 * @param { AssignedExperiments | null } storedExperiments - The experiment data currently stored in IndexedDB.
```
parse the new data from the fetch, get the old one (indexDb), merge both and prepare for storage again. I will change or re-write a better description for this.